File: python/Problem022.py
def score(name):
    return sum([ord(letter)-64 for letter in name])

f = open("../problemdata/names.txt","r")
filedata = f.read()
f.close()

names = filedata.split(",")

cleaned =  [ bob.strip('"') for bob in names]
cleaned.sort()

total = 0
for i, name in enumerate(cleaned):
    total += score(name)*(i+1)

print total

File: golang/Problem022.go
package main

import (
	"euler"
	"fmt"
	"sort"
	"strings"
	"time"
)

func score(name string) (total int) {
	for _, let := range name {
		total += int(let - 64)
	}
	return total
}

func main() {
	starttime := time.Now()

	data := euler.Import("../problemdata/names.txt")
	names := strings.Split(data[0], ",")
	for i, name := range names {
		names[i] = strings.Trim(name, "\"")
	}

	sort.Strings(names)

	total := 0
	for i, name := range names {
		total += (i + 1) * score(name)
	}

	fmt.Println(total)

	fmt.Println("Elapsed time:", time.Since(starttime))
}
